neaten/neaten_test.go

```diff
@@ -0,0 +1,328 @@
+/*
+******************************************************************************
+* Copyright (c) 2022 Genome Research Ltd.
+*
+* Author: Sendu Bala <sb10@sanger.ac.uk>
+*         Kyle Mace <km34@sanger.ac.uk>
+*
+* Permission is hereby granted, free of charge, to any person obtaining
+* a copy of this software and associated documentation files (the
+* "Software"), to deal in the Software without restriction, including
+* without limitation the rights to use, copy, modify, merge, publish,
+* distribute, sublicense, and/or sell copies of the Software, and to
+* permit persons to whom the Software is furnished to do so, subject to
+* the following conditions:
+*
+* The above copyright notice and this permission notice shall be included
+* in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*****************************************************************************
+ */
+package neaten
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const modePermUser = 0700
+
+func TestTidy(t *testing.T) { //nolint:gocognit
+	date := "20220829"
+	srcUniversal := "cci4fafnu1ia052l75sg"
+	srcUniqueGo := "cci4fafnu1ia052l75t0"
+	srcUniquePerl := "cci4fafnu1ia052l75tg"
+
+	Convey("Given existing source and dest dirs you can tidy the source", t, func() {
+		tmpDir := t.TempDir()
+		srcDir := filepath.Join(tmpDir, "src", srcUniversal)
+		destDir := filepath.Join(tmpDir, "dest", "final")
+		interestUniqueDir1 := createTestPath([]string{srcDir, "go", srcUniqueGo})
+		interestUniqueDir2 := createTestPath([]string{srcDir, "perl", srcUniquePerl})
+
+		buildSrcDir(srcDir, srcUniqueGo, srcUniquePerl, interestUniqueDir1, interestUniqueDir2)
+
+		createTestDirWithDifferentPerms(destDir)
+
+		combineSuffixes := map[string]string{
```

Comment:
It makes more sense if all of these properties are returned by createTestDirWithDifferentPerms(), since it is that function that is creating files with these names (is choosing the names).

Reply:
I don't think line 57 is creating files with anything in combineSuffixes? But it is creating a dest dir, so I have changed it so that it returns this as a string variable